src/main/java/filter/expression/Qualifier.java

```diff
@@ -1031,11 +1031,9 @@ private Qualifier convertMilestoneAliasQualifier(List<TurboMilestone> allMilesto
             }
         }
 
-        // clamp
-        if (currentIndex >= allMilestones.size()) {
-            currentIndex = allMilestones.size() - 1;
-        } else if (currentIndex < 0) {
-            currentIndex = 0;
+        // if out of milestone range, don't convert alias
+        if (currentIndex >= allMilestones.size() || currentIndex < 0) {
+            return this;
```

Comment:
Is that a good idea? It seems unlikely that that is what the user would have intended.